beetsplug/fetchart.py

```diff
@@ -310,6 +310,24 @@ def get(self, album, plugin, paths):
                 match=Candidate.MATCH_FALLBACK)
 
 
+class CoverArtArchiveReleaseGroup(RemoteArtSource):
+    NAME = u"Cover Art Archive Release Group"
+
+    if util.SNI_SUPPORTED:
+        URL = 'https://coverartarchive.org/release-group/{mbid}/front'
+    else:
+        URL = 'http://coverartarchive.org/release-group/{mbid}/front'
+
+    def get(self, album, plugin, paths):
+        """Return the Cover Art Archive release group URLs using album
+        MusicBrainz release group ID.
+        """
+        if album.mb_releasegroupid:
+            yield self._candidate(
+                    url=self.URL.format(mbid=album.mb_releasegroupid),
+                    match=Candidate.MATCH_FALLBACK)
```

Comment:
Would you mind doing a little refactoring to help avoid duplicating some code? For example, a base `CoverArtArchive` class and two subclasses, `CAARelease` and `CAAReleaseGroup`, might help share this code and some of the `SNI_SUPPORTED` logic for choosing the URL. That will make maintenance of this plugin easier in the future.

Reply:
Thank you for the feedback! I made the changes you suggested. I was a little unsure where to write the documentation for the feature since I could not find any documentation on the other fetchart sources. Please, come with suggestions on improvements.